File: app/src/main/java/com/myflightbook/android/FlightQueryActivity.java
/*
	MyFlightbook for Android - provides native access to MyFlightbook
	pilot's logbook
    Copyright (C) 2017-2019 MyFlightbook, LLC

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.myflightbook.android;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;

public class FlightQueryActivity extends AppCompatActivity {

    private ActFlightQuery m_flightQuery;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.fragmenthost);

        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        m_flightQuery = new ActFlightQuery();
        fragmentTransaction.replace(android.R.id.content, m_flightQuery);
        fragmentTransaction.commitAllowingStateLoss();
    }

    @Override
    public void onBackPressed() {
        Intent mIntent = new Intent();
        mIntent.putExtra(ActFlightQuery.QUERY_TO_EDIT, m_flightQuery.getCurrentQuery());
        setResult(Activity.RESULT_OK, mIntent);
        finish();
        super.onBackPressed();
    }

}

File: app/src/main/java/com/myflightbook/android/FlightQueryActivity.kt
/*
	MyFlightbook for Android - provides native access to MyFlightbook
	pilot's logbook
    Copyright (C) 2017-2020 MyFlightbook, LLC

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.myflightbook.android

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.content.Intent

class FlightQueryActivity : AppCompatActivity() {
    private var mFlightquery: ActFlightQuery? = null
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.fragmenthost)
        val fragmentManager = supportFragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()
        mFlightquery = ActFlightQuery()
        fragmentTransaction.replace(android.R.id.content, mFlightquery!!)
        fragmentTransaction.commitAllowingStateLoss()
    }

    override fun onBackPressed() {
        val mIntent = Intent()
        mIntent.putExtra(ActFlightQuery.QUERY_TO_EDIT, mFlightquery!!.getCurrentQuery())
        setResult(RESULT_OK, mIntent)
        finish()
        super.onBackPressed()
    }
}